SpatialGDK/Source/SpatialGDK/Public/Schema/StandardLibrary.h

```diff
@@ -55,9 +55,9 @@ inline void AddCoordinateToSchema(Schema_Object* Object, Schema_FieldId Id, cons
 	Schema_AddDouble(CoordsObject, 3, Coordinate.Z);
 }
 
-inline Coordinates GetCoordinateFromSchema(Schema_Object* Object, Schema_FieldId Id)
+inline Coordinates IndexCoordinateFromSchema(Schema_Object* Object, Schema_FieldId Id, uint32 Index)
```

Comment:
Looking at the c_worker.h
```
/* Functions that access a value by index for a particular field ID in an object. Boolean values are
 * guaranteed to be either 0 or 1.
 *
 * If the index doesn't exist for the given field, a default is returned; call
 * the corresponding GetCount function above to to determine if the total number of fields.
 *
 * Note that, for best performance, fields should be accessed in field ID and index order. */
```
I'm wondering about error handling here if an incorrect `Index` is used. I believe we should either do a `GetCount` before creating the `Schema_IndexObject` or check if the result is this `default`.

Reply:
Then again, perhaps it's up to the caller of `IndexCoordinateFromSchema` to call `GetCount` before trying to index.... I'll go with that. Could we add a comment informing users that this is the recommended flow?

Reply:
From what I can tell via uses of the various other `IndexYFromSchema` functions, I think there's an implicit expectation that:
a) The function is going to be invoked from within a loop that provides the `Index` argument and
b) The Index argument will exist in the range `[0, GetYCount...)`

Do we want to do a documentation pass over _all_ of the `IndexYFromSchema` functions noting that this is the recommended usage? Would it be worth doing that as a separate PR?